src/mapping/config/MappingConfiguration.cpp

```diff
@@ -24,134 +25,169 @@
   }
-  // Add tags that only, but all RBF mappings use
-  for (XMLTag &tag : tags) {
-    tag.addAttribute(attrSolverRtol);
-    tag.addAttribute(attrPolynomial);
-    tag.addAttribute(attrPreallocation);
-    tag.addAttribute(attrXDead);
-    tag.addAttribute(attrYDead);
-    tag.addAttribute(attrZDead);
-    tag.addAttribute(attrUseLU);
+  case BasisFunctions::VolumeSplines: {
+    return PtrMapping(new typename BackendSelector<T, VolumeSplines>::type(constraint, dimension, mapping::VolumeSplines(), std::forward<Args>(args)...));
   }
-  {
-    XMLTag tag(*this, VALUE_NEAREST_NEIGHBOR, occ, TAG);
-    tag.setDocumentation("Nearest-neighbour mapping which uses a rstar-spacial index tree to index meshes and run nearest-neighbour queries.");
-    tags.push_back(tag);
+  case BasisFunctions::Multiquadrics: {
+    return PtrMapping(new typename BackendSelector<T, Multiquadrics>::type(constraint, dimension, mapping::Multiquadrics(shapeParameter), std::forward<Args>(args)...));
   }
-  {
-    XMLTag tag(*this, VALUE_NEAREST_PROJECTION, occ, TAG);
-    tag.setDocumentation("Nearest-projection mapping which uses a rstar-spacial index tree to index meshes and locate the nearest projections.");
-    tags.push_back(tag);
+  case BasisFunctions::InverseMultiquadrics: {
+    return PtrMapping(new typename BackendSelector<T, InverseMultiquadrics>::type(constraint, dimension, mapping::InverseMultiquadrics(shapeParameter), std::forward<Args>(args)...));
   }
-  {
-    XMLTag tag(*this, VALUE_NEAREST_NEIGHBOR_GRADIENT, occ, TAG);
-    tag.setDocumentation("Nearest-neighbor-gradient mapping which uses nearest-neighbor mapping with an additional linear approximation using gradient data.");
-    tags.push_back(tag);
+  case BasisFunctions::Gaussian: {
+    return PtrMapping(new typename BackendSelector<T, Gaussian>::type(constraint, dimension, mapping::Gaussian(shapeParameter), std::forward<Args>(args)...));
   }
-  {
-    XMLTag tag(*this, VALUE_LINEAR_CELL_INTERPOLATION, occ, TAG);
-    tag.setDocumentation("Linear cell interpolation mapping which uses a rstar-spacial index tree to index meshes and locate the nearest cell. Only supports 2D meshes.");
-    tags.push_back(tag);
+  default:
+    PRECICE_UNREACHABLE("No instantiation was found for the selected basis function.");
+    return nullptr;
   }
+}
 
+} // namespace
+
+MappingConfiguration::MappingConfiguration(
+    xml::XMLTag &              parent,
+    mesh::PtrMeshConfiguration meshConfiguration)
+    : _meshConfig(std::move(meshConfiguration))
+{
+  PRECICE_ASSERT(_meshConfig);
+  using namespace xml;
+
+  // First, we create the available tags
+  XMLTag::Occurrence occ = XMLTag::OCCUR_ARBITRARY;
+  std::list<XMLTag>  projectionTags, rbfDirectTags, rbfIterativeTags, rbfAliasTag;
+  createTag(*this, TYPE_NEAREST_NEIGHBOR, occ, TAG, projectionTags, "Nearest-neighbour mapping which uses a rstar-spacial index tree to index meshes and run nearest-neighbour queries.");
```

Comment:
Is there a reason why we don't create the tags directly inside the storage constructor?

```cpp
std::list<XMLTag> projectionTags{
  XMLTag{*this, TYPE_NEAREST_PROJECTION, occ, TAG}
    .setDocumentation("Nearest-neighbor-gradient mapping which uses nearest-neighbor mapping with an additional linear approximation using gradient data."),
  XMLTag{*this, TYPE_NEAREST_NEIGHBOR_GRADIENT, occ, TAG}
    .setDocumentation("Nearest-neighbor-gradient mapping which uses nearest-neighbor mapping with an additional linear approximation using gradient data."),
// etc
};
```

Reply:
In see your point (no extra function), but I would probably prefer the current version of the code as it is less error-prone with respect to the brackets and commas, better readable due to less indentations in the code and I don't have to think about initializer-lists for each list.
However, if that's not convincing enough for you I will change everything to an initializer list.

Reply:
The only thing I don't like is the extra function, which doesn't add a lot of functionality and makes it even easier to pass the arguments in the wrong order.

If the initializer list is an issue, then you can also use individual push_backs:
```cpp
std::list<XMLTag> projectionTags;
projectionTags.push_back(XMLTag{*this, TYPE_NEAREST_PROJECTION, occ, TAG}
    .setDocumentation("Nearest-neighbor-gradient mapping which uses nearest-neighbor mapping with an additional linear approximation using gradient data."));
projectionTags.push_back(XMLTag{*this, TYPE_NEAREST_NEIGHBOR_GRADIENT, occ, TAG}
    .setDocumentation("Nearest-neighbor-gradient mapping which uses nearest-neighbor mapping with an additional linear approximation using gradient data."));
```

clang-format should deal with the braces etc. and make it quick and easy to catch mistakes.

Reply:
Done